### ISSUE TITLE
New version 2014/11/27

### ISSUE THREAD

**Issue by ViperCode7** [delan/facepaste#22]
I've fixed and improved the 2.8 official version and here's the changelog:
- fixed locations for both user and business pages
- fixed albums not loading on some pages
- fixed user name not being found on album page
- improved interface by adding default selections to options
- improved interface by adding default path (that of default Download dir, IMPORTANT: such a dir must be selected in Firefox options)
- unfixed bug: there still are some minor cosmetic issues when default page address is just /photos, that Photos is listed as being Photos of and not Photos by or vice versa depending on what the user has as default for Photos
- unfixed bug: if the user has no Photos, but just the Albums tab the proper procedures won't be loaded, the user has to click on Albums title link in the page to reload that page but with proper address in order to make it work.

Other than that, as far as I was able to test it, it works fully.

Download link, drag and drop on the main folder, it should override any old facepaste versions:
https://www.dropbox.com/s/wf8jfbwwd74a3i7/facepaste-2.9-fx.xpi?dl=0

Please report bugs for my fixed version of facepaste here, on this thread, not on the main forum.

**Comment by akostadinov** [comment]
@emm980, would you post your source code somewhere so that your changes can be merged back into the project? In the `xpi` file I see a lot of js merged into a single file.

**Comment by emm980** [comment]
Right now I don't have spare time to add this but to make the ext work on pages where the url is not in form of facebook.com/name but is in form of facebook.com/page/name see my reply to josh5671, there is the change needed to make it work.

Once I have some time I'll rewrite the URL test part to make it work properly, also got some idea to fix those bugs I mentioned in original post and hopefully make this URL business go away for good.

**Comment by Eraser099** [comment]
It does work someway, but I can notice a problem related to photo album browsing, both on Firefox and Chrome, after I do use of Facepaste: even after a few minutes of pics download session from the fb album pages, at my subsequent log in (on fb), I can see pages conceal the access to the single albums and/or the actual number of pictures stored in every single album (this happened after the last download session). Maybe fb servers consider the facepaste batch download as a form of hacking and then, as a consequence, try to prevent it downloading pics?

**Comment by b3bis** [comment]
I don't believe it is considered a hack, it is considered a batch download (as it is indeed), something which by the TOS is not really what you would consider the most acceptable thing taking into consideration that generally (if I can recall well enough) the use of software which does not use the FB api is not acceptable. Anyway, me too had seen some minor problems while and after using the extension. They may have implemented some form of image uri requests per second, though not something far from this.

**Comment by Eraser099** [comment]
It looks like changes about the number of pics contained in every single album is a new permanent change, since a few people I know have been experiencing the same thing. According to your words, batch download of pics is not considered the most acceptable thing... you think people who does batch download, not more than once per month, can be persecuted or at least banned from their servers? I guess admins are aware about who's the one who's downloading...

**Comment by josh5671** [comment]
Facebook seen change something again?..

These few weeks batch download all stuck on downloading (lime green color block)

**Comment by TrueTenacity** [comment]
Not sure if it was FB or firefox that changed...

I was downloading like a demon yesterday, this morning I updated firefox
and now also getting the stuck on downloading...

**Comment by josh5671** [comment]
@TrueTenacity Maybe I try rollback some old version firefox and take a test

**Comment by Xaemyl** [comment]
Yeah its firefox 35-36 that broke it. I rolled back to firefox 34.0.5 for the time being and it works again (although it does stall on pages with lots of albums on it, so have to download a few albums at a time).

**Comment by DavePlater** [comment]
Ok. I think I fixed the "stuck downloading" bug. The problem seemed to be that progressListener no longer triggers the "i am done" message. The first 8 files download but that is it.
So i took emm980 's version and switched it over to using the Downloads.jsm 
Also added some quick options for how many photos to download at a time (was hardcoded at 8 before)
At any rate, it works in FF 36.0.4 (my version requires FF 26 or better)

**Download link, drag and drop on the main folder, it should override any old facepaste versions:**
https://www.dropbox.com/s/dffibvvw4lc01nv/facepaste-2.9b-fx.xpi?dl=0

**Comment by TrueTenacity** [comment]
Hi Dave,

I tried your new version and it doesn't seem to work... might be because of
me being behind a proxy or something... but its not downloading the album
list or letting me browse to pick a folder to download to.

FF 35.0.1 / Windows 7 x64

**Comment by maciex** [comment]
I does not work for me either. It also stuck on downloading the album list.

FF 36.0.1

**Comment by DavePlater** [comment]
Well shoot.
When it fails to load the album list, does the cancel button "not work" and you have to close with the X ?
That means the javascript is broken.
For a quick check, go to the Add-ons Manager (Ctrl+Shift+A i think?) and go to the Options for facepase. Check and uncheck the box and change the number setting.
I suspect the default values are NOT being set and the javascript breaks because of it.

**Comment by TrueTenacity** [comment]
You are correct, have to use the X...

I changed the settings and it seems to be working now...

**Comment by maciex** [comment]
After changing the settings the "Cancel" button still doesn't seem to work. Also the plugin itself doesn't download the album list.

**Comment by DavePlater** [comment]
The settings bug was because i changed a setting name but didnt fix it in the defaults, re-uploaded with a fix for THAT problem.
**maciex** are you on a page that would find the albums with emm980's version?
*www.facebook.com/{user-page}/photos   ??

**Comment by maciex** [comment]
@DavePlater: I now installed the reuploaded version from dropbox and tried on page https://www.facebook.com/aksienionek.szymon/photos

I still got the error in Browser console:
NS_ERROR_FAILURE: Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIProperties.get] facepaste.js:28:0

**Comment by DavePlater** [comment]
Hmm, best I can figure your system doesn't have the properties/settings service ready by the time I call it. Probably my fault for not doing some kind of "wait" to change when it loads in the lifecycle.
I can probably just repackage an "option-less" version that has the download fix and none of my extra goodies.
What OS are you on?
Also, how did you find that error? I've been pretty much doing this all blind with no ability to see errors or output (which is why it took me way longer than it should)

**Comment by maciex** [comment]
W dniu 27.03.2015 o 14:42, DavePlater pisze:
> Hmm, best I can figure your system doesn't have the properties/settings
> service ready by the time I call it. Probably my fault for not doing
> some kind of "wait" to change when it loads in the lifecycle.
> I can probably just repackage an "option-less" version that has the
> download fix and none of my extra goodies.

That would be nice, as a quick fix. I think the default values are fine 
for me.

> What OS are you on?

GNU Linux (Arch) x86_64 Kernel 3.19.1-1-ck

> Also, how did you find that error? I've been pretty much doing this all
> blind with no ability to see errors or output (which is why it took me
> way longer than it should)

I used Browser Console. It will show you all JavaScript errors.
https://developer.mozilla.org/en-US/docs/Tools/Browser_Console

-- 
Pozdrawiam
-- 
Maciej Sitarz

**Comment by DavePlater** [comment]
I edited my original post to have the option-less version as well, but here it is again:
https://www.dropbox.com/s/pd1pocbme5985g3/facepaste-2.9b-nop-fx.xpi?dl=0
Hopefully that one will work

**Comment by interceptorchamp** [comment]
Hello,
I installed your newest version however I can´t seem to get it work.
I´m using Firefox 36 on Win7 64.
Like the previous versions, it gets stuck when trying to download 8 pics in lime green color.
The size of these files on my HDD is 0 KB.

Can you picture out what is wrong?
The previous version up to approx. 6 weeks ago worked fine.

I recommend that you put a link on the main page to make it easier to find this update page.

Greetings from Germany

**Comment by emm980** [comment]
You could look up handle_photo_page and see what's not working. I got img null on "photos by" page. It seems neither fbPhotosPhotoActionsItem nor .fbPhotoImage is there thus p.photourl always gets to nothing.

Getting r.response.body content all I got was This content is currently unavailable.
![err](https://cloud.githubusercontent.com/assets/6399179/6882078/4df191d2-d582-11e4-83f5-3111d871d464.png)

Also it's strange that !link && !img gets passed... in the case above I was expecting to see that p.log( 'error... pop up.

The first release, with options, did not work, until I applied the changes to the options, then it downloaded several albums completely with no issue.

Now, on a different user page it seems it can't get that p.photourl...

It's very annoying when fb uses several versions of their software to push pages, all working differently.

Here are two plain users, one is working (pandor.ioana, has fbPhotosPhotoActionsItem in r.response) and the other one is not working under any circumstance (miha.mihaela.56).

**Comment by DavePlater** [comment]
Using:
Windows7 32Bit
Locale: en-US
Firefox: 37.0.1
And FP: 2.9b-nop

I downloaded the albums from (miha.mihaela.56) and (pandor.ioana) just fine.
However, (aksienionek.szymon) found the album but claimed there were no photos inside to download.

**Comment by interceptorchamp** [comment]
I tried to download an album from both facebook users, and both stuck in lime green after 8 files, each of 0 Bytes length :(
The folders are created correct.

Any ideas what I could do to get it fixed?

**Comment by DavePlater** [comment]
@interceptorchamp If you are using the plain 2.9 version at the top of this thread, it will stall out like that. Look for my two links to versions 2.9b and 2.9b-nop

**Comment by josh5671** [comment]
I think the 2.9 work well in newer version firefox.
but somehow the pages album are not listed in album.
only 'Photo by...'are shown

**Comment by Xaemyl** [comment]
Also 2.9 doesnt seem to be able to download pages videos anymore.

**Comment by ViperCode7** [comment]
The way facebook versions pages based on language/location makes it difficult to track what does and what doesn't work.

It's a good project, if fb hadn't removed ways in their API to obtain certain data... now mostly addons might work on single pages where the whole data is already set by the page itself. FB Album mod still works alright, it's not a complete solution, but it should be able to counter changes and limitation.

**Comment by interceptorchamp** [comment]
I am having the same issue with video pages.
Pictures run fine.
Videos are not being downloaded anymore, but it worked some months ago.
Can you figure out what was changed by facebook?

**Comment by interceptorchamp** [comment]
Hello,

during the last few days, facepaste shows after fetching the album index for some time "found 0 photos" and terminates the execution, although my LAN Gadget shows some traffic in both directions (upload and download).

This happens in different user profiles, independently of the geographical Location.

Do you know of recent changes of Facebook which causes the described issues?

I would appreciate your help to make facepast work again.

**Comment by josh5671** [comment]
Anyone going to update the code, 
which this few month facebook update their code...

**Comment by DavePlater** [comment]
I've been working on it, will post a new thread with what I've got.
I don't think it will fix everyone's issues. But it fixed the ones I was having.

**Comment by interceptorchamp** [comment]
Great!
I appreciate you effort in keeping us updated :-)(
I´m looking Forward to Keep it a try ...

**Comment by b3bis** [comment]
I 've been watching the modifications and done one or two myself but now it does not seem to work at all. It gets to a 500 error for every photo (albums are detected just fine). I could help with the programming part but I know almost nothing about ff extensions. 

Also on firefox add on page there is a facepaste-reload addon that claims to be what is done here but doesn't seem to incorporate the video inclusion changes. Also trying that add-on I got the same results but with an error asking if I have 3rd party cookies enabled (which I have) if this is of any help.

Also, shouldn't we try to combine forces with whoever updates that extension?
